astropy/io/ascii/ui.py

```diff
@@ -271,6 +272,16 @@ def _guess(table, read_kwargs, format, fast_reader):
             # user supplies delimiter="|" but the guess wants to try delimiter=" ",
             # so skip the guess entirely.
             continue
+
+        if guess_kwargs not in filtered_guess_kwargs:
```

Comment:
This line eliminates duplicate guesses.  It turns out in some cases the `basic` format could have 3 or 4 repeats.